tests/bdd/common/helm.py

```diff
@@ -0,0 +1,186 @@
+            check=True,
+            text=True,
+        )
+        return repo_name
+
+    except subprocess.CalledProcessError as e:
+        print(
+            f"Error: command 'helm repo add mayastor https://openebs.github.io/mayastor-extensions' failed with exit code {e.returncode}"
+        )
+        print(f"Error Output: {e.stderr}")
+        return None
+
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
+        return None
+
+
+class ChartSource(Enum):
+    HOSTED = "mayastor"
+    LOCAL = local_chart_dir()
+
+
+class HelmReleaseClient:
+    """
+    A client for interacting with Helm releases in a specified Kubernetes namespace.
+
+    Attributes:
+        namespace (str): The Kubernetes namespace where the Helm releases are managed.
+        storage_driver (str): The Helm storage driver to use.
+    """
+
+    def __init__(self, namespace: str, storage_driver: str = ""):
+        """
+        Initializes the HelmReleaseClient.
+
+        Args:
+            namespace (str): The Kubernetes namespace where the Helm releases are managed.
+            storage_driver (str): The Helm storage driver to use. Defaults to "".
+        """
+        self.namespace = namespace
+        self.storage_driver = storage_driver
+
+    def list(self):
+        """
+        Lists the deployed Helm releases in the specified namespace.
+
+        Executes the 'helm ls' command to retrieve a list of deployed releases.
+
+        Returns:
+            str: A newline-separated string of deployed release names, or None if an error occurs.
+        """
+        try:
+            result = subprocess.run(
+                ["helm", "ls", "-n", self.namespace, "--deployed", "--short"],
+                capture_output=True,
+                check=True,
+                text=True,
+                env={"HELM_DRIVER": self.storage_driver},
+            )
+            return result.stdout.strip()
+
+        except subprocess.CalledProcessError as e:
+            print(
+                f"Error: command 'helm ls -n {self.namespace} --deployed --short' failed with exit code {e.returncode}"
+            )
+            print(f"Error Output: {e.stderr}")
+            return None
+
+        except Exception as e:
+            print(f"An unexpected error occurred: {e}")
+            return None
+
+    def release_is_deployed(self, release_name: str):
+        releases = self.list()
+        if releases is not None:
+            for release in releases:
+                if release == release_name:
+                    return True
+        return False
+
+    def install_mayastor(self, source: ChartSource, version=None):
+        if self.release_is_deployed("mayastor"):
+            print(
+                f"WARN: Helm release 'mayastor' already exists in the 'mayastor' namespace."
+            )
+            return
+
+        install_command = []
+        if source == ChartSource.HOSTED:
+            repo_name = repo_add_mayastor()
```

Comment:
ah ok, we want to be able to do both, got it.
Let's change the script to support this as well, rather than have separate commands here

Reply:
I'll put this comment in the tests PR 564.